transifex/native/django/management/commands/try_templatetag.py

```diff
@@ -0,0 +1,361 @@
+            context_value) in itertools.product(
+                tag_names, sources, source_filters, params, param_filters,
+                asvars, asvar_filters, blocks, context_values
+    ):
+        try:
+            test = make_test(tag_name, source, source_filter, param,
+                             param_filter, asvar, asvar_filter, block,
+                             context_value)
+        except TypeError:
+            pass
+        else:
+            results.append(test)
+
+    # Remove duplicates
+    results = [(template, json.dumps(context))
+               for template, context in results]
+    results = sorted(set(results))
+    results = [(template, json.loads(context))
+               for template, context in results]
+    return results
+
+
+def make_test(tag_name, source=None, source_filter=None, param=None,
+              param_filter=None, asvar=None, asvar_filter=None, block=None,
+              context_value=None):
+
+    """ Compose a template and context for testing based on the arguments.  """
+
+    if not source and not block:
+        raise TypeError("At least one of 'source' or 'block' must be "
+                        "specified")
+
+    result = ['{% ', tag_name, ' ']
+    context_vars = []
+    if source:
+        if source[0] not in ('"', "'"):
+            context_vars.append(source)
+        for match in re.findall(r'\{[\w_]+\}', source):
+            context_vars.append(match[1:-1])
+        result.append(source)
+    if source_filter:
+        result.extend(['|', source_filter])
+    if param:
+        context_vars.append(param)
+        result.extend([' ', param, '=', param])
+        if param_filter:
+            result.extend(['|', param_filter])
+    if asvar:
+        result.extend([' as ', asvar])
+    result.append(' %}')
+    if not source and block:
+        for match in re.findall(r'\{[\w_]+\}', block):
+            context_vars.append(match[1:-1])
+        result.extend([block, '{% end', tag_name, ' %}'])
+    if asvar:
+        result.extend(['{{ ', asvar])
+        if asvar_filter:
+            result.extend(['|', asvar_filter])
+        result.extend([' }}'])
+
+    context = {}
+    if context_value is not None:
+        context = {context_var: context_value for context_var in context_vars}
+
+    return ''.join(result), context
+
+
+def test(template_str, context_dict=None, autoescape=True, i=''):
+    """ Use the django templating engine to run a test.
+
+        Arguments:
+
+        template_str: The template to render
+        context_dict: The context to render the template against
+        autoescape:   Pretend the django templating engine was setup with
+                      autoescape or not (in most real use-cases, it will have
+                      been set up with autoescape=True)
+        lang_code:    The language to translate to
+        i:            Prepend the output with this in order to help distinguish
+                      tests when multiple are run
```

Comment:
Can we keep the same formatting in docstring params throughout the files? (`:param ...`)